### PR TITLE
Adding support for Pow instances in mrv function

### PR DESCRIPTION
**Pull request by anutosh491**
Towards #2526

### CODE REVIEW

File: src/libasr/pass/replace_symbolic.cpp [hunk 1]
```diff
@@ -985,7 +997,9 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
                     ASRUtils::EXPR(ASR::make_Var_t(al, x.base.base.loc, symbol))));
             }
             symbolic_vars_to_free.clear();
-            pass_result.push_back(al, ASRUtils::STMT(ASR::make_Return_t(al, x.base.base.loc)));
+            for (ASR::symbol_t* symbol : symbolic_vars_to_omit) {
+                symbolic_vars_to_free.insert(symbol);
+            }
```
Comment (certik) [inline src/libasr/pass/replace_symbolic.cpp:1002]: This code right here makes https://github.com/lcompilers/lpython/issues/2552 fail.

File: src/libasr/pass/replace_symbolic.cpp [hunk 2]
```diff
@@ -1007,6 +1007,18 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
         }
     }
 
+    void visit_WhileLoop(const ASR::WhileLoop_t &x) {
+        ASR::WhileLoop_t &xx = const_cast<ASR::WhileLoop_t&>(x);
+        transform_stmts(xx.m_body, xx.n_body);
+        if (ASR::is_a<ASR::IntrinsicScalarFunction_t>(*xx.m_test)) {
+            ASR::IntrinsicScalarFunction_t* intrinsic_func = ASR::down_cast<ASR::IntrinsicScalarFunction_t>(xx.m_test);
+            if (intrinsic_func->m_type->type == ASR::ttypeType::Logical) {
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/replace_symbolic.cpp:1015]: Minor
```suggestion
            if (ASRUtils::is_logical(intrinsic_func->m_type)) {
```
Or
```suggestion
            if (is_a<ASR::Logical_t>(*intrinsic_func->m_type)) {
```

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Comment by Thirumalai-Shaktivel** [comment]
Update the branch and squash merge this.

**Comment by anutosh491** [comment]
Thanks for the review

**Pull request merged**